src/pointcloud_io.jl

```diff
@@ -530,76 +586,106 @@ function save_las(filename::AbstractString, pc::AbstractVector{<:NamedTuple}; x_
+                                point_return_count[min(p.returnnumber, length(point_return_count))] += 1
+                            else
+                                point_return_count[1] += 1
+                            end
+                        end
+                    else
+                        data_format_id = 1
+                        data_record_length = 28
+                        data = Vector{LasPoint1}(undef, records_count)
+                        @inbounds for i in 1:records_count
+                            p = pc[i]
+                            data[i] = laspoint1(p, x_offset, y_offset, z_offset, x_scale, y_scale, z_scale)
+                            if haskey(p, :returnnumber)
+                                point_return_count[p.returnnumber] += 1
+                            else
+                                point_return_count[1] += 1
+                            end
+                        end
+                    end
+                end
+            else
+                if haskey(first(pc), :color)
+                    data_format_id = 2
+                    data_record_length = 26
+                    data = Vector{LasPoint2}(undef, records_count)
                     @inbounds for i in 1:records_count
                         p = pc[i]
-                        data[i] = laspoint7(p, x_offset, y_offset, z_offset, x_scale, y_scale, z_scale)
+                        data[i] = laspoint2(p, x_offset, y_offset, z_offset, x_scale, y_scale, z_scale)
                         if haskey(p, :returnnumber)
-                            point_return_count[min(p.returnnumber, length(point_return_count))] += 1
+                            point_return_count[p.returnnumber] += 1
                         else
                             point_return_count[1] += 1
                         end
                     end
                 else
-                    data_format_id = 3
-                    data_record_length = 34
-                    data = Vector{LasPoint3}(undef, records_count)
+                    data_format_id = 0
+                    data_record_length = 20
+                    data = Vector{LasPoint0}(undef, records_count)
                     @inbounds for i in 1:records_count
                         p = pc[i]
-                        data[i] = laspoint3(p, x_offset, y_offset, z_offset, x_scale, y_scale, z_scale)
+                        data[i] = laspoint0(p, x_offset, y_offset, z_offset, x_scale, y_scale, z_scale)
                         if haskey(p, :returnnumber)
                             point_return_count[p.returnnumber] += 1
                         else
                             point_return_count[1] += 1
                         end
                     end
                 end
-            else
-                data_format_id = 1
-                data_record_length = 28
-                data = Vector{LasPoint1}(undef, records_count)
```

Comment:
I don't know why you are changing the earlier formats of 0->3 and haven't checked if it's right, not sure it matters thoes would be old files now

Reply:
I added some code and git thinks I have changed it but it only has been pushed around.